experimental/oak_baremetal_launcher/src/instance_binary.rs

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::{instance::LaunchedInstance, BinaryParams};
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use log::info;
+use std::{
+    net::Shutdown,
+    os::unix::net::{UnixListener, UnixStream},
+};
+
+/// An instance of the runtime running directly as a linux binary
+pub struct BinaryInstance {
+    tempdir: Option<tempfile::TempDir>,
+    comms_host: UnixStream,
+    instance: tokio::process::Child,
+}
+
+impl BinaryInstance {
+    pub fn start(params: BinaryParams) -> Result<Self> {
+        let tempdir = tempfile::tempdir().context("failed to create temp dir")?;
```

Comment:
I don't think there's any point in giving the socket a name, just use raw file descriptors.

Use `UnixStream::pair()` to get a pair of anonymous connected sockets, pass one to the binary and use the other yourself, and you avoid having to deal with the temporary directories (and possibly leaving them around when you crash).

(As a side note, when implementing the original qemu/crosvm loaders, I thought about using tempdir as well, but tried really hard to avoid having to give a name to the sockets.)

Reply:
The downside to raw file descriptors is that it would require us to use unsafe rust in the loader to turn the file descriptor provided in the args into a UnixSocket. 

We can do it if you strongly feel that's the better way, but I thought all things considered temp files are probs fine

Reply:
Which particular unsafe function would be used for the conversion? I don't think it's a problem to have unsafe code, and I also think it would look cleaner than the temp files.

Reply:
std::os::unix::io::FromRawFd
https://doc.rust-lang.org/std/os/unix/io/trait.FromRawFd.html

Reply:
> This function is also unsafe as the primitives currently returned have the contract that they are the sole owner of the file descriptor they are wrapping. Usage of this function could accidentally allow violating this contract which can cause memory unsafety in code that relies on it being true.

Reply:
Inside the loader you should be able to just use `/dev/fd/<x>` as a regular file, no need for unsafe code.

Reply:
updated to use a file descriptor. We did end up having to use unsafe code, but should be fine